### PR TITLE
Add identity server integration

### LINKED ISSUE #3: Integrate WingedKeys/IdentityServer
Allow for authenticated sign in with WingedKeys

### PR DESCRIPTION
**Pull request by cailyncodes**
Should close #3

### CODE REVIEW

File: client/src/contexts/AuthenticationContext/AuthenticationContext.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+		fetchServiceConfiguration();
+	}, [openIdConnectUrl]);
+
+	// Make an authorization request whenever:
+	// 1) the authorizationHandler is set,
+	// 2) the configuration is set or changes, and
+	// 3) it is the login screen (isLogin or isCallback)
+	useEffect(() => {
+		if (configuration && authorizationHandler) {
+			if (isLogin) {
```
Comment (jlhogan) [inline client/src/contexts/AuthenticationContext/AuthenticationContext.tsx:177]: can this be cleaned up at all by making more 'pure functions' as we discussed in some other ticket in the other repo somewhere?

File: docker-compose.yaml [hunk 2]
```diff
@@ -22,11 +22,34 @@ services:
...
+
+  winged-keys-db:
+    image: mcr.microsoft.com/mssql/server
+    environment:
+      SA_PASSWORD: TestPassword1
+      ACCEPT_EULA: Y
+  winged-keys:
+    build:
+        context: .
+        dockerfile: ./winged-keys/Dockerfile
```
Comment (jlhogan) [inline docker-compose.yaml:42]: can we add checkout of winged keys repo requirement to README?

File: client/src/contexts/AuthenticationContext/AuthenticationContext.tsx (hunk 1 above)
Comment (cailyncodes) [reply client/src/contexts/AuthenticationContext/AuthenticationContext.tsx:177]: This is the same code from `Hedwig`. Open to discussions on purifying things but don't think this PR is the one for that.

Comment (jlhogan) [reply client/src/contexts/AuthenticationContext/AuthenticationContext.tsx:177]: my thinking is always that the most likely time a non essential change is gonna get made is _now_ hehe but that's fine

**Pull request merged**